yarn-project/archiver/src/archiver/archiver_store.ts

```diff
@@ -310,39 +288,23 @@ export class MemoryArchiverStore implements ArchiverDataStore {
   }
 
   /**
-   * Gets the `take` amount of encrypted logs starting from `from`.
-   * @param from - Number of the L2 block to which corresponds the first encrypted logs to be returned.
-   * @param take - The number of encrypted logs to return.
-   * @returns The requested encrypted logs.
-   */
-  public getEncryptedLogs(from: number, take: number): Promise<L2BlockL2Logs[]> {
-    if (from < INITIAL_L2_BLOCK_NUM) {
-      throw new Error(`Invalid block range ${from}`);
-    }
-    if (from > this.encryptedLogs.length) {
-      return Promise.resolve([]);
-    }
-    const startIndex = from - INITIAL_L2_BLOCK_NUM;
-    const endIndex = from + take;
-    return Promise.resolve(this.encryptedLogs.slice(startIndex, endIndex));
-  }
-
-  /**
-   * Gets the 'take' amount of unencrypted logs starting from 'from'.
-   * @param from - Number of the L2 block to which corresponds the first unencrypted logs to be returned.
-   * @param take - The number of unencrypted logs to return.
-   * @returns The requested unencrypted logs.
+   * Gets the `take` amount of logs starting from `from`.
+   * @param from - Number of the L2 block to which corresponds the first logs to be returned.
+   * @param take - The number of logs to return.
+   * @param logType - Specifies whether to return encrypted or unencrypted logs.
+   * @returns The requested logs.
    */
-  public getUnencryptedLogs(from: number, take: number): Promise<L2BlockL2Logs[]> {
+  getLogs(from: number, take: number, logType: 'encrypted' | 'unencrypted'): Promise<L2BlockL2Logs[]> {
```

Comment:
I wonder if an enum type would feel more natural here rather than strings?

Reply:
Good idea. Addressed in [cfaa9cb](https://github.com/AztecProtocol/aztec-packages/pull/929/commits/cfaa9cb4b95f515722eb64c091e206c8eae25c7d)